hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/web3j/TestWeb3jService.java

```diff
@@ -0,0 +1,302 @@
+import org.web3j.protocol.core.Response;
+import org.web3j.protocol.core.methods.request.Transaction;
+import org.web3j.protocol.core.methods.response.EthCall;
+import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
+import org.web3j.protocol.core.methods.response.EthGetTransactionReceipt;
+import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.protocol.websocket.events.Notification;
+import org.web3j.tx.Contract;
+import org.web3j.tx.gas.ContractGasProvider;
+import org.web3j.tx.gas.DefaultGasProvider;
+import org.web3j.utils.Numeric;
+
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class TestWeb3jService implements Web3jService {
+
+    private static final Long GAS_LIMIT = 15_000_000L;
+    private static final String MOCK_KEY = "0x4e3c5c727f3f4b8f8e8a8fe7e032cf78b8693a2b711e682da1d3a26a6a3b58b6";
+
+    private final ContractExecutionService contractExecutionService;
+    private final ContractGasProvider contractGasProvider;
+    private final Credentials credentials;
+    private final DomainBuilder domainBuilder;
+    private final Map<String, String> trxResMap = new HashMap<>();
+    private final Web3j web3j;
+
+    private Address sender = Address.fromHexString("");
+
+    public TestWeb3jService(ContractExecutionService contractExecutionService, DomainBuilder domainBuilder) {
+        this.contractExecutionService = contractExecutionService;
+        this.contractGasProvider = new DefaultGasProvider();
+        this.credentials = Credentials.create(ECKeyPair.create(Numeric.hexStringToByteArray(MOCK_KEY)));
+        this.domainBuilder = domainBuilder;
+        this.web3j = Web3j.build(this);
+    }
+
+    public void setSender(Address sender) {
+        this.sender = sender;
+    }
+
+    public void setSender(String sender) {
+        this.sender = Address.fromHexString(sender);
+    }
+
+    @SneakyThrows(Exception.class)
+    public <T extends Contract> T deploy(Deployer<T> deployer) {
+        return deployer.deploy(web3j, credentials, contractGasProvider).send();
+    }
+
+    @Override
+    public <T extends Response> T send(Request request, Class<T> responseType) {
+        final var method = request.getMethod();
+        return switch (method) {
+            case "eth_call" -> (T) ethCall(request.getParams(), request);
+            case "eth_getTransactionCount" -> (T) ethGetTransactionCount();
+            case "eth_getTransactionReceipt" -> (T) getTransactionReceipt(request);
+            case "eth_sendRawTransaction" -> (T) call(request.getParams(), request);
+            default -> throw new UnsupportedOperationException(request.getMethod());
+        };
+    }
+
+    private EthSendTransaction call(List<?> params, Request request) {
+        var rawTransaction = TransactionDecoder.decode(params.get(0).toString());
+        var trxHex = generateTransactionHashHexEncoded(rawTransaction, credentials);
```

Comment:
Rename this `trxHex` everywhere else too.

```suggestion
        var transactionHash = generateTransactionHashHexEncoded(rawTransaction, credentials);
```

Reply:
Renamed